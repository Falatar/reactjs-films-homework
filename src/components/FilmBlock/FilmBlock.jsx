import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FilmBlockFrontSide from '../FilmBlockFrontSide';
import FilmBlockBackSide from '../FilmBlockBackSide';

class FilmBlock extends Component {
  constructor(props) {
    super(props);
    this.state = {
      detailsMode: false,
      poster: '',
    };
  }

  componentDidMount() {
    const { imgURL } = this.props;
    this.setImg(imgURL);
  }

  switchMode = () => {
    this.setState((state) => ({
      detailsMode: !state.detailsMode,
    }));
  }

  callModal = () => {
    const { callTrailer, setTrailer, id } = this.props;
    setTrailer(id);
    callTrailer();
  }

  checkImgSizes = () => {
    if (this.width + this.height === 0) {
      this.onerror();
    }
  }

  setDefaultImg = () => {
    this.setState({ poster: 'https://cdn.pixabay.com/photo/2016/11/30/12/16/question-mark-1872665_1280.jpg' });
  }

  setImg = (imgURL) => {
    this.setState({ poster: `https://image.tmdb.org/t/p/w500${imgURL}` });
  }

  render() {
    const {
      id, name, tagList, rating, overview,
    } = this.props;
    const { detailsMode, poster } = this.state;
    if (!detailsMode) {
      return (
        <FilmBlockFrontSide
          id={id}
          name={name}
          tagList={tagList}
          poster={poster}
          rating={rating}
          callModal={this.callModal}
          switchMode={this.switchMode}
          checkImgSizes={this.checkImgSizes}
          setDefaultImg={this.setDefaultImg}
        />
      );
    }
    return (
      <FilmBlockBackSide
        name={name}
        tagList={tagList}
        poster={poster}
        rating={rating}
        overview={overview}
        callModal={this.callModal}
        switchMode={this.switchMode}
      />
    );
  }
}

FilmBlock.defaultProps = {
  id: 0,
  name: 'Can\'t find property "name"',
  imgURL: '',
  rating: 0,
  tagList: 'Can\'t find property "tagList"',
  overview: 'Can\'t find property "overview"',
  callTrailer: () => 'can\'t find callTrailer function',
  setTrailer: () => 'can\'t find setTrailer function',
};

FilmBlock.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  imgURL: PropTypes.string,
  rating: PropTypes.number,
  tagList: PropTypes.string,
  overview: PropTypes.string,
  callTrailer: PropTypes.func,
  setTrailer: PropTypes.func,
};

export default FilmBlock;
