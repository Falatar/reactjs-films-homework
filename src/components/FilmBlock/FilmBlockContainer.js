import { connect } from 'react-redux';
import { updateTrailerPath, openModal } from '../../modules/singleFilmModule/singleFilmModuleActions';
import FilmBlock from './FilmBlock';


export const mapDispatchToProps = (dispatch) => ({
  callTrailer: () => dispatch(openModal()),
  setTrailer: (path) => dispatch(updateTrailerPath(path)),
});

export default connect(
  null,
  mapDispatchToProps,
)(FilmBlock);
