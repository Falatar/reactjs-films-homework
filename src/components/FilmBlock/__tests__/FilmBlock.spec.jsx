import React from 'react';
import renderer from 'react-test-renderer';
import FilmBlock from '../FilmBlock';

jest.mock('../../FilmBlockFrontSide', () => 'FilmBlockFrontSide');
jest.mock('../../FilmBlockBackSide', () => 'FilmBlockBackSide');

describe('FilmBlock', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<FilmBlock imgURL="" id={1} name="nemo" tagList="empty" rating={0.0} overview="some text" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
