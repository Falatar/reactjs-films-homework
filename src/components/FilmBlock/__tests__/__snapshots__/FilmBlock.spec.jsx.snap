// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmBlock renders correctly 1`] = `
<FilmBlockFrontSide
  callModal={[Function]}
  checkImgSizes={[Function]}
  id={1}
  name="nemo"
  poster="https://image.tmdb.org/t/p/w500"
  rating={0}
  setDefaultImg={[Function]}
  switchMode={[Function]}
  tagList="empty"
/>
`;
